FinSearchUnified/Tests/Helper/UrlBuilderTest.php

```diff
@@ -150,4 +168,165 @@ private function setIpHeader($field, $ipAddress)
+
+        $httpResponse = new Zend_Http_Response(200, [], $responseBody);
+
+        $this->httpClient->expects($this->once())
+            ->method('request')
+            ->willReturn($httpResponse);
+
+        $urlBuilder = new UrlBuilder($this->httpClient);
+
+        $status = $urlBuilder->getConfigStatus();
+
+        $this->assertEquals(
+            $expectedStatus,
+            $status,
+            'Expected integration type to be "Direct Integration"'
+        );
+    }
+
+    /**
+     * @dataProvider unsuccessfulRequestProvider
+     *
+     * @param string $integrationType
+     * @param bool $expectedStatus
+     *
+     * @throws Zend_Http_Exception
+     * @throws Exception
+     */
+    public function testIntegrationTypeOnUnsuccessfulRequest($integrationType, $expectedStatus)
+    {
+        $configArray = [
+            ['IntegrationType', $integrationType]
+        ];
+
+        // Create Mock object for Shopware Config
+        $config = $this->getMockBuilder(Shopware_Components_Config::class)
+            ->setMethods(['offsetGet'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $config->expects($this->atLeastOnce())
+            ->method('offsetGet')
+            ->willReturnMap($configArray);
+
+        // Assign mocked config variable to application container
+        Shopware()->Container()->set('config', $config);
+
+        $httpResponse = new Zend_Http_Response(500, [], '');
+
+        $this->httpClient->expects($this->once())
+            ->method('request')
+            ->willReturn($httpResponse);
+
+        $urlBuilder = new UrlBuilder($this->httpClient);
+
+        $status = $urlBuilder->getConfigStatus();
+
+        $this->assertEquals(
+            $expectedStatus,
+            $status,
+            sprintf(
+                'Expected integration type to be "%s"',
+                $integrationType
+            )
+        );
+    }
+
+    /**
+     * @dataProvider exceptionOnRequestProvider
+     *
+     * @param string $integrationType
+     * @param bool $expectedStatus
+     *
+     * @throws Exception
+     */
+    public function testIntegrationTypeOnException($integrationType, $expectedStatus)
+    {
+        $configArray = [
+            ['IntegrationType', $integrationType]
+        ];
+
+        // Create Mock object for Shopware Config
+        $config = $this->getMockBuilder(Shopware_Components_Config::class)
+            ->setMethods(['offsetGet'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $config->expects($this->atLeastOnce())
+            ->method('offsetGet')
+            ->willReturnMap($configArray);
+
+        // Assign mocked config variable to application container
+        Shopware()->Container()->set('config', $config);
+
+        $this->expectException(Exception::class);
+
+        $this->httpClient->expects($this->once())
+            ->method('request')
+            ->willThrowException(new Exception());
```

Comment:
Use `Zend_Http_Client_Exception` instead. This will work once `$this->expectException(Exception::class);` has been removed.